lib/src/tar/tar_file.dart

```diff
@@ -153,18 +151,18 @@ class TarFile {
 
     final headerBytes = header.getBytes();
 
-    // The checksum is calculated by taking the sum of the unsigned byte values
-    // of the header record with the eight checksum bytes taken to be ascii
-    // spaces (decimal value 32). It is stored as a six digit octal number
-    // with leading zeroes followed by a NUL and then a space.
+    /// The [checksum] is calculated by taking the sum of the unsigned byte
+    /// values of the header record with the eight checksum bytes taken to be
+    /// ascii spaces (decimal value 32). It is stored as a six digit octal
+    /// number with leading zeroes followed by a NULL and then a space.
     var sum = 0;
     for (var b in headerBytes) {
       sum += b;
     }
 
     var sum_str = sum.toRadixString(8); // octal basis
     while (sum_str.length < 6) {
-      sum_str = '0' + sum_str;
+      sum_str += '0';
```

Comment:
Did you switch to an append instead of a prepend for a reason? The loop isn't great, but it should still prepend the 0 padding.